Whollet/Whollet/Source/Presentation/Detail/DetailViewModel.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import Charts
+
+enum TimeChart {
+    case day
+    case week
+    case month
+    case year
+}
+
+final class DetailViewModel: NSObject {
+    private var getCoinsUseCase: GetCoinsUseCase!
+    private var getChartUseCase: GetChartlUseCase!
+    
+    private(set) var bsCoin: BehaviorSubject<CoinModel> = BehaviorSubject(nil)
+    private(set) var bsChart: BehaviorSubject<[ChartDataEntry]> = BehaviorSubject(nil)
+    
+    override init() {
+        super.init()
+        self.getCoinsUseCase = GetCoinsUseCase()
+        self.getChartUseCase = GetChartlUseCase()
+    }
+    
+    func getDetail(id: String) {
+        let param = CoinsRequestParams(
+            order: "market_cap_desc",
+            perPage: 1,
+            page: 1,
+            sparkline: false,
+            ids: id)
+        
+        self.getCoinsUseCase.getCoins(params: param) { [weak self] (data, error) -> Void in
+            guard let self = self, error == nil else { return }
+            if let data = data, data.count == 1 {
+                self.bsCoin.add(data.last)
+            }
```

Comment:
e break line từ điều kiện nhé
```suggestion
            guard let self = self,   error == nil,  let data = data, data.count == 1 else { return }
            self.bsCoin.add(data.last)
```